src/osd/PrimaryLogPG.cc

```diff
@@ -317,6 +321,13 @@ class CopyFromCallback : public PrimaryLogPG::CopyCallback {
       }
       ctx->pg->close_op_ctx(ctx);
     }
+
+    // Only use truncate_{seq,size} from the original object if the client
+    // did not sent us these parameters
+    if (!truncate_seq && !truncate_size) {
```

Comment:
I'm not entirely sure about this — aren't zeroes valid numbers here?
@ukernel, I see client/Inode.h initializes to 1 by default, but the OSD and MDS both start at 0.

So I think we need a separate indicator if they're set or not.

Reply:
Ok, instead of adding yet another magic value to these 2 parameters, I've just sorted it out by adding an extra setter to the CopyFromCallback class which will keep track whether the truncate_{seq,size} parameters have been set by the client or not.

I've also changed the OSD_COPY_FILE feature definition so that it doens't overlap NAUTILUS.  This way, we can backport the feature and check it on the client.

(and sorry for the delay in my reply)

Reply:
👍